pagopa/io-app
# feat: [IOCOM-870] New DS on payment list in the SEND message details

## Short description
This PR adds the payments section on a SEND message details, with the new DS for both standard and cancelled message.
Payment button and payments bottom sheet will come in another PR.
|No payments|One payment|
|-|-|
|![Simulator Screenshot - iPhone 15 - 2024-03-29 at 12 11 58](https://github.com/pagopa/io-app/assets/5150343/66167f83-e14a-470f-9047-0dea1a03eaec)|![Simulator Screenshot - iPhone 15 - 2024-03-29 at 12 12 08](https://github.com/pagopa/io-app/assets/5150343/1a4f4c68-39c5-4eb7-98ae-f5b665ba8762)|

|Less than five payments|More than five payments| 
|-|-|
|![Simulator Screenshot - iPhone 15 - 2024-03-29 at 12 12 31](https://github.com/pagopa/io-app/assets/5150343/d0392415-93b9-4a74-aeec-6dafe9b445a0)|![Simulator Screenshot - iPhone 15 - 2024-03-29 at 12 12 45](https://github.com/pagopa/io-app/assets/5150343/4e29e40f-1451-4e5f-b7d4-3f9ca181a381)|

|Cancelled<br/>no payments|Cancelled<br/>at least one unpaid payment|
|-|-|
|![Simulator Screenshot - iPhone 15 - 2024-03-29 at 12 02 25](https://github.com/pagopa/io-app/assets/5150343/b46c946c-6460-44ce-954a-a43e24e1c9ec)|![Simulator Screenshot - iPhone 15 - 2024-03-29 at 12 02 36](https://github.com/pagopa/io-app/assets/5150343/6a1a4eeb-b1fc-425c-9544-d856c893f971)|

|Cancelled<br/>one paid payment|Cancelled<br/>more than one paid payment|
|-|-|
|![Simulator Screenshot - iPhone 15 - 2024-03-29 at 12 04 09](https://github.com/pagopa/io-app/assets/5150343/62c86cd7-1499-4d59-97d3-f334d5fbee92)|![Simulator Screenshot - iPhone 15 - 2024-03-29 at 12 04 23](https://github.com/pagopa/io-app/assets/5150343/9dc3497a-4354-4939-99e7-8078cd6a3af8)|

## List of changes proposed in this pull request
- New `MessagePayments` component that displays the payment list (even for a cancelled SEND message)
- Old component renamed to `LegacyMessagePayments`

## How to test
Using the io-dev-api-server, generate some SEND messages and check that the new payment component behaves properly. Also check that the payment flow work as expected.
 Suggested configuration:
- standard SEND message with no payments
- standard SEND message with one payment
- standard SEND message with five payments
- standard SEND message with more than five payments
- cancelled SEND message with no payments
- cancelled SEND message with one or more unpaid/failed payments
- cancelled SEND message with a mix of paid and unpaid/failed payments
